src/Infrastructure/Captcha/GoogleRecaptchaV3Service.cs

```diff
@@ -0,0 +1,135 @@
+﻿using Microsoft.Extensions.Configuration;
+using System.Runtime.Serialization.Json;
+using System.Web;
+
+namespace FSH.WebApi.Infrastructure.Captcha;
+
+public class GoogleRecaptchaV3Service : ICaptchaService
+{
+    public HttpClient _httpClient { get; set; }
+
+    public string RequestPath { get; set; }
+    public string RequestSecret { get; set; }
+    public string RequestResponse { get; set; }
+    public string RequestRemoteip { get; set; }
+
+    public GResponseModel Response { get; set; }
+
+    public HttpRequestException HttpReqException { get; set; }
+
+    public Exception GeneralException { get; set; }
+
+
+    public GoogleRecaptchaV3Service(HttpClient httpClient, IConfiguration config)
+    {
+        _httpClient = httpClient;
+        var settings = config.GetSection(nameof(CaptchaSettings)).Get<CaptchaSettings>();
+
+        RequestSecret = settings.GoogleRecaptchaV3.Secret;
+        RequestPath = settings.GoogleRecaptchaV3.ApiUrl;
+    }
+
+    public void InitializeRequest(string RecaptchaToken, string RemoteIpAddress)
+    {
+        RequestRemoteip = RemoteIpAddress;
+        RequestResponse = RecaptchaToken;
+    }
+
+    public async Task<bool> Execute()
+    {
+        // Notes on error handling:
+        // Google will pass back a 200 Status Ok response if no network or server errors occur.
+        // If there are errors in on the "business" level, they will be coded in an array;
+        // CaptchaRequestException is for these types of errors.
+
+        // CaptchaRequestException and multiple catches are used to help seperate the concerns of 
+        //  a) an HttpRequest 400+ status code
+        //  b) an error at the "business" level 
+        //  c) an unpredicted error that can only be handled generically.
+
+        // It might be worthwhile to implement a "user error message" property in this class so the
+        // calling procedure can decide what, if anything besides a server error, to return to the 
+        // client and any client handling from there on.
+        try
+        {
+
+            // Don't to forget to invoke any loggers in the logic below.
+
+            // formulate request
+            string builtURL = $"{RequestPath}?" +
+                HttpUtility.UrlPathEncode($"secret={RequestSecret}&response={RequestResponse}&remoteip={RequestRemoteip}");
+            StringContent content = new StringContent(builtURL);
+
+            // send request, await.
+            HttpResponseMessage response = await _httpClient.PostAsync(builtURL, null);
+            response.EnsureSuccessStatusCode();
+
+            // read response
+            byte[] res = await response.Content.ReadAsByteArrayAsync();
+
+            string logres = await response.Content.ReadAsStringAsync();
+            Console.WriteLine($"Retrieved Response: {logres}");
```

Comment:
> Can I give you a general advice: don't blindly copy code from the internet.
> 
> You did a good effort, don't get me wrong... Just don't think you picked the right article. ;-)
> 
> It would also be nice to see the client side implemented on the blazor wasm project. So we can see it in action.
> 
> I see you added it to the tokens service now. I think the most important place where we need a captcha is the new user registration. I wouldn't even make it optional there, at least when Anonymous is allowed.
> 
> So we'll probably need a separate switch "enableOnLogin", or multiple if there are more places where a captcha would be appropriate.
> 
> We should also design the interface so it can be easily implemented for other captcha providers. Maybe we should support 2 different providers from the start so we can make the abstraction generic enough and ensure we don't leak implementation details...



> Can I give you a general advice: don't blindly copy code from the internet.
> 
> You did a good effort, don't get me wrong... Just don't think you picked the right article. ;-)
> 
> It would also be nice to see the client side implemented on the blazor wasm project. So we can see it in action.
> 
> I see you added it to the tokens service now. I think the most important place where we need a captcha is the new user registration. I wouldn't even make it optional there, at least when Anonymous is allowed.
> 
> So we'll probably need a separate switch "enableOnLogin", or multiple if there are more places where a captcha would be appropriate.
> 
> We should also design the interface so it can be easily implemented for other captcha providers. Maybe we should support 2 different providers from the start so we can make the abstraction generic enough and ensure we don't leak implementation details...


No absolute wrong or right... just implementation... 
Thanks for review ...

Yes interface also options may be more general/abstract !

Also today i noticed service implementation is wrong 
if you chose dont use capcha option... (conditional injection required really adds complexity)
becouse of this
i guess is more convience that 
making service part static and use if capcha enabled way 

so get this as draft or idea ...

no worries about word.. i am not getting personal... 
just trying to add value to this solution am much as have time.... 
issue is really dont have time to give more.
this impl. took only appox. one hour..